llvm/tools/sycl-post-link/ModuleSplitter.cpp

```diff
@@ -761,5 +761,138 @@ getDoubleGRFSplitter(ModuleDesc &&MD, bool EmitOnlyKernelsAsEntryPoints) {
     return std::make_unique<ModuleCopier>(std::move(MD), std::move(Groups));
 }
 
+namespace {
+  struct KernelProperties {
```

Comment:
nit: this struct works with aspects, so maybe the name could reflect that rather than 'properties'

Reply:
The idea is to expand it further to also handle things like `reqd_sub_group_size`, `large_gfr`, etc. So, the intent of is to be some abstract properties. Perhaps we can come up with a different name like "device requirements" or "optional features"

Reply:
"Features" seem nice

Reply:
Renamed to `UsedOptionalFeatures` in 18ba12c8fc05aa6facfe4e630aaac75e11f405d9